src/linear_algebra/eigenspace.lean

```diff
@@ -325,6 +325,38 @@ lemma generalized_eigenspace_eq_generalized_eigenspace_findim_of_le
   f.generalized_eigenspace μ k = f.generalized_eigenspace μ (findim K V) :=
 ker_pow_eq_ker_pow_findim_of_le hk
 
+theorem aeval_apply_of_has_eigenvector [field K] [vector_space K V] {f : End K V}
+  {p : polynomial K} {μ : K} {x : V} (h : f.has_eigenvector μ x) :
+  aeval f p x = (p.eval μ) • x :=
+begin
+  apply p.induction_on,
+  { intro a, simp [module.algebra_map_End_apply] },
+  { intros p q hp hq, simp [hp, hq, add_smul] },
+  { intros n a hna,
+    rw [mul_comm, pow_succ, mul_assoc, alg_hom.map_mul, linear_map.mul_app, mul_comm, hna],
+    simp [algebra_map_End_apply, mem_eigenspace_iff.1 h.2, smul_smul, mul_comm] }
+end
+
+section minimal_polynomial
+
+variables [field K] [vector_space K V] [finite_dimensional K V] (f : End K V)
+
+theorem is_integral : is_integral K f :=
+is_integral_of_noetherian (by apply_instance) f
+
+theorem is_root_of_has_eigenvalue (μ : K) (h : f.has_eigenvalue μ) :
+  (minimal_polynomial f.is_integral).is_root μ :=
+begin
+  rcases (submodule.ne_bot_iff _).1 h with ⟨w, ⟨H, ne0⟩⟩,
+  have foo := @smul_eq_zero _ _ _ _ _ (eval μ (minimal_polynomial f.is_integral)) w,
+  rw [← aeval_apply_of_has_eigenvector ⟨ne0, H⟩, minimal_polynomial.aeval f.is_integral] at foo,
+  simp only [true_iff, eq_self_iff_true, linear_map.zero_apply] at foo,
+  cases foo, { rwa is_root },
+  exfalso, exact ne0 foo,
```

Comment:
I'm assuming `foo` is a temporary name? Hopefully we can find some way to hide the mathematically trivial logical manipulations inside this proof.